### ISSUE TITLE
[BUG] Validations fail for the following crons

### ISSUE THREAD

**Issue by dsalaza4** [grumBit/aws_cron_expression_validator#6]
**Describe the bug**

Running validations on the following crons return exceptions:

- `0 8 * ? 2-6 *`
- `0 11-23/2 * * ? *`
- `0 11-23/4 ? * 2-6 *`
- `0 11-23/2 * * ? *`
- `0 0 1 1-12/3 ? *`

**To Reproduce**

Run `AWSCronExpressionValidator.validate` for any of the provided crons.

**Expected behaviour**

All provided crons should be valid

**Additional context**

I am using `aws-cron-expression-validator: "1.0.11"`

**Comment by dsalaza4** [comment]
Maybe it is the `number-number` format? 

`2-6`, `11-23/2`, `11-23/4`, `11-23/2` and `1-12/3` ?

**Comment by grumBit** [comment]
Hi @dsalaza4,

Thank you for finding those problems and letting me know.

Except for the first one, I think you are right, it should be possible to precede a `/` with a range.  I think in your second example, the `11-23/2` in the `Hour` field means "every second hour between 11 and 23" (i.e. 11, 13, 15, 17, 19 & 23). I've updated the code and added your examples as test cases. If you could try the latest version, 1.1.2, that would be great please.


For the first example, `0 8 * ? 2-6 *`, I think the validator is correctly identifying it as an invalid expression. I'm wondering if you have the third and fourth fields swapped around?

```
┌───────────── 0   Minute
│ ┌─────────── 8   Hour
│ │ ┌───────── *   Day-of-month
│ │ │ ┌─────── ?   Month
│ │ │ │  ┌──── 2-6 Day-of-week
│ │ │ │  │  ┌─ *   Year
0 8 * ? 2-6 *
```

The exception says `Invalid combination of day-of-month '*' and day-of-week '2-6'. One must be a question mark (?)`, which I think matches [the AWS notes](https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-create-rule-schedule.html#eb-cron-expressions); _"You can't specify the Day-of-month and Day-of-week fields in the same cron expression. If you specify a value or a * (asterisk) in one of the fields, you must use a ? (question mark) in the other."_. Perhaps you have the `*` in the `Day-of-month` field swapped with the `?` in the `Month` field?

Many thanks,
Grum.

**Comment by dsalaza4** [comment]
Hey @grumBit,

Thank you so much for such an amazing response!

The first cron was indeed incorrect.

I just tested all the provided cases again and everything went just fine.

Thank you!

**Comment by grumBit** [comment]
Fantastic! Thanks @dsalaza4. Let me know if anything else crops up 😃

**Comment by grumBit** [comment]
Hi  @dsalaza4, 

It may not affect you, however, after fixing the bug you found, [I found another one](https://github.com/grumBit/aws_cron_expression_validator/issues/7), which is fixed in v1.1.3. So, you might want to upgrade to it.

Cheers!

**Comment by dsalaza4** [comment]
@grumBit Thank you so much for the heads up!
